### PR TITLE
Bug/msp 9972/appscan import

### PR DESCRIPTION
**Pull request by thelightcosine**
This fixes customer reported bugs affecting the database imports for IBM Rational AppScan XML files. With certain XML files, importing would trigger a stack trace. This uncovered multiple bugs in the AppScan Nokogiri parser which are fixed herein. 

This PR must be reviewed by an internal team member as verification requires the use of specific XML files which are not public.
## PRO VERIFICATION STEPS
- [x] start Metasploit Pro
- [x] create a new workspace
- [x] download the AppScan.zip attached to the Jira ticket
- [x] extract the xml file from the zip
- [x] in Metasploit Pro go to Import and select the AppScan.xml
- [x] VERIFY the import completes successfully
- [x] Also verify with QE test import file
## MSF VERIFICATION STEPS
- [x] `./msfconsole`
- [x] Connect to a DB, new workspace
- [x] `db_import AppScan.xml`
- [ ] Verify objects in workspace

### CODE REVIEW

**Comment by jlee-r7** [comment]
This should almost certainly also be tested against `db_import` in msfconsole.

**Comment by shuckins-r7** [comment]
@dmaloney-r7 The QE test file (attached to Jira as demo.xml) imported without error, data looked good. The AppScan.xml from that zip, however, hit an error, full stack in ticket.

File: lib/rex/parser/appscan_nokogiri.rb [hunk 1]
```diff
@@ -187,31 +187,18 @@ def report_service_info
...
-        request_body_text = split_traffic[1].to_s[0,content_length]
-      else
-        request_body_text = nil
-      end
-      response_headers_text = split_traffic[1].to_s[content_length,split_traffic[1].to_s.size].lstrip
-      request = request_headers_text
-      return unless(request && response_headers_text)
-      response_body_text = split_traffic[2]
+      request_headers, request_body, response_headers, response_body = really_original_traffic.split(/\r\n\r\n/)
+      return unless(request_headers && response_headers)
```
Comment (shuckins-r7) [inline lib/rex/parser/appscan_nokogiri.rb]: request_body and response_body are presumed to be non-nil below, so they should also be guarded here.

**Comment by shuckins-r7** [comment]
Web proof issue still present: https://github.com/rapid7/metasploit-framework/issues/5034

**Pull request merged**